### PR TITLE
add a link to store.metabase.com for hosted instances

### PR DESCRIPTION
**Pull request by kdoh**
Closes https://github.com/metabase/harbormaster/issues/1139

## What this does
- Adds an item to the user profile dropdown if the instance is hosted to make it easy to get back to store.metabase.com to manage instance details.

### CODE REVIEW

**Review by mazameli — APPROVED** [review]

**Comment by kdoh** [comment]
@nemanjaglumac I'm a bit confused by the test failure here. Any insight?

**Comment by nemanjaglumac** [comment]
> @nemanjaglumac I'm a bit confused by the test failure here. Any insight?

I'll take a look and try to figure it out.

**Comment by nemanjaglumac** [comment]
@kdoh Does this help at all?

`at Settings.isHosted (frontend/src/********/lib/settings.js:91:34)`

```javascript
// Right now, all Metabase Cloud hosted instances run on *.metabaseapp.com
// We plan on changing this to look at an envvar in the future instead.
isHosted() {
  return this.get("site-url")
    .toLowerCase()
    .includes("metabaseapp.com");
}
```

**Comment by mazameli** [comment]
I think it's mad that the site url might be null. I pushed something that I think will fix it but we'll see.

**Comment by nemanjaglumac** [comment]
> I think it's mad that the site url might be null. I pushed something that I think will fix it but we'll see.

Unit test was failing because `ProfileLink` now calls `isHosted()` method that is checking for `site-url`. Unit test in its original form didn't handle hosted instances. I've adjusted that test by mocking `isHosted()` in https://github.com/metabase/metabase/pull/14511/commits/e71fa59639a8dadf03733aea7db94a5f6851685f. Now we have it covered for every non-hosted instance, but I'd like to include additional checks to accommodate this newest change.

**Comment by nemanjaglumac** [comment]
@kdoh and @mazameli - test now checks against hosted instances, as well. (https://github.com/metabase/metabase/pull/14511/commits/df195b76f918036d96676896d17fcf407d46a567)
I wanted to keep it dry, but please let me know if this is too abstract or if it's hard to follow/read? 

My part is now done. Feel free to take over and proceed with the merge if everything else is in place. 🚀

Also, keep an eye this warning: https://github.com/metabase/metabase/pull/14511/checks?check_run_id=1765779818.

**Comment by mazameli** [comment]
@salsakran, kdoh had tagged you for a review, but all this does is simply adds a link to `https://store.metabase.com/account` in the top-right dropdown of the app when `isHosted` is true. Seem reasonable? I'd like to get this merged 'cause it adds a new string and I'm trying to batch all the last straggling strings together so I can do a last run through the whole POEditor auto translate flow for 0.38.

image

File: frontend/src/metabase/lib/settings.js [hunk 1]
```diff
@@ -88,9 +88,12 @@ class Settings {
...
   isHosted() {
-    return this.get("site-url")
-      .toLowerCase()
-      .includes("metabaseapp.com");
+    return (
+      this.get("site-url") &&
+      this.get("site-url")
+        .toLowerCase()
+        .includes("metabaseapp.com")
+    );
```
Comment (nemanjaglumac) [inline frontend/src/metabase/lib/settings.js:96]:
```suggestion
    return /.+@metabase.com$/i.test(this.get("site-url"));
```

- This takes care of the case.
- If no url or if it's undefined, this will return `false` (no need for an additional check)
- More importantly, it makes sure that the we allow **only** `metabase.com` domain (addresses warning [Incomplete URL substring sanitization](https://github.com/metabase/metabase/pull/14511/checks?check_run_id=1765933701))

Comment (nemanjaglumac) [reply frontend/src/metabase/lib/settings.js:96]: If you find one-line solution too terse:

```javascript
isHosted() {
  const regex = new RegExp(/.+@metabase.com$/i)
  return regex.test(this.get("site-url"));
}
```

Comment (flamber) [reply frontend/src/metabase/lib/settings.js:96]: The domain used for hosting is `<custom>.metabaseapp.com` - not `metabase.com` and it's not an email.

Comment (nemanjaglumac) [reply frontend/src/metabase/lib/settings.js:96]: Oops, yes. I was too tired last night :D
Thank you

File: frontend/src/metabase/nav/components/ProfileLink.jsx [hunk 2]
```diff
@@ -45,6 +45,15 @@ export default class ProfileLink extends Component {
         link: Urls.accountSettings(),
         event: `Navbar;Profile Dropdown;Edit Profile`,
       },
+      ...(MetabaseSettings.isHosted() &&
+        admin && [
+          {
+            title: t`Manage hosting`,
```
Comment (kdoh) [inline frontend/src/metabase/nav/components/ProfileLink.jsx:51]:
```suggestion
            title: t`Manage Metabase Cloud`,
```

File: frontend/test/metabase/nav/ProfileLink.unit.spec.js [hunk 3]
```diff
@@ -1,49 +1,69 @@
...
 
+const REGULAR_ITEMS = [
+  "Account settings",
+  "Activity",
+  "Help",
+  "About Metabase",
+  "Sign out",
+];
+const ADMIN_ITEMS = [...REGULAR_ITEMS, "Admin"];
+const HOSTED_ITEMS = [...ADMIN_ITEMS, "Manage hosting"];
```
Comment (kdoh) [inline frontend/test/metabase/nav/ProfileLink.unit.spec.js:15]:
```suggestion
const HOSTED_ITEMS = [...ADMIN_ITEMS, "Manage Metabase Cloud"];
```
